### PR TITLE
Tracing configuration json marshal/unmarshal

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Needed for passing the configuration from the adapter controller to the receive adapter https://github.com/knative/eventing/issues/2978

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: tracing/config/tracing.go [hunk 1]
```diff
@@ -139,3 +140,43 @@ func NewTracingConfigFromMap(cfgMap map[string]string) (*Config, error) {
...
+	out := make(map[string]string, 5)
+	out[backendKey] = string(cfg.Backend)
+	if cfg.ZipkinEndpoint != "" {
+		out[zipkinEndpointKey] = cfg.ZipkinEndpoint
+	}
+	if cfg.StackdriverProjectID != "" {
+		out[stackdriverProjectIDKey] = cfg.StackdriverProjectID
+	}
+	out[debugKey] = strconv.FormatBool(cfg.Debug)
+	out[sampleRateKey] = strconv.FormatFloat(cfg.SampleRate, 'f', -1, 64)
```
Comment (vagababov) [inline tracing/config/tracing.go:178]: I find `Sprint`  easier to read, especially for the float case.
```suggestion
	out[debugKey] = fmt.Sprint(cfg.Debug)
	out[sampleRateKey] = fmt.Sprint(cfg.SampleRate)
```

File: tracing/config/tracing_test.go [hunk 2]
```diff
@@ -137,6 +137,102 @@ func TestNewConfigSuccess(t *testing.T) {
...
+			}
+
+			haveConfig, err := JsonToTracingConfig(json)
+			if err != nil {
+				t.Fatal("Failed to create tracing config:", err)
+			}
+
+			if diff := cmp.Diff(tc.output, haveConfig); diff != "" {
+				t.Errorf("Got config from map (-want, +got) = %v", diff)
+			}
```
Comment (vagababov) [inline tracing/config/tracing_test.go:231]:
```suggestion
			if !cmp.Equal(tc.output, haveConfig) {
				t.Error("Got config from map (-want, +got) =", cmp.Diff(tc.output, haveConfig))
			}
```

**Comment by matzew** [comment]
@n3wscott looks like your `/Approve` needs to be `/approve` :rofl:

**Pull request merged**